src/test/java/com/bookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,182 @@
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ShoppingCartControllerTest {
+    private static final Long VALID_ID = 1L;
+    private static final Long USER_ID = 2L;
+    private static final String TITLE = "Colony";
+    private static final int VALID_QUANTITY = 5;
+    private static final String ADD_USER_CART_AND_CART_ITEMS =
+            "classpath:database/cart&cartitems/add_user_cart_and_cartitems.sql";
+    private static final String DELETE_USER_CART_AND_CART_ITEMS =
+            "classpath:database/cart&cartitems/delete_user_cart_and_cartitems.sql";
+    private static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext webApplicationContext) {
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @DisplayName("Valid case of the getShoppingCart method run with valid request")
+    @Sql(scripts = ADD_USER_CART_AND_CART_ITEMS,
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = DELETE_USER_CART_AND_CART_ITEMS,
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void getShoppingCart_validRequest_isOk() throws Exception {
+        String token = jwtUtil.generateToken("user@example.com");
+
+        ShoppingCartResponseDto expected = buildValidCartResponseDto();
+        MvcResult result = mockMvc.perform(get("/cart")
+                        .header(HttpHeaders.AUTHORIZATION, "Bearer " + token)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        ShoppingCartResponseDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), ShoppingCartResponseDto.class);
+
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+    }
```

Comment:
If I am not mistaken, you can use @WithMockUser and specify username as email of existing in your db user. Frankly, it's a comfortable way for testing such cases
```suggestion
    @WithMockUser(username = USER_NAME, roles = ...)
    @Test
    @DisplayName("Valid case of the getShoppingCart method run with valid request")
    @Sql(scripts = ADD_USER_CART_AND_CART_ITEMS,
            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
    @Sql(scripts = DELETE_USER_CART_AND_CART_ITEMS,
            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
    void getShoppingCart_validRequest_isOk() throws Exception {
        ShoppingCartResponseDto expected = buildValidCartResponseDto();
        MvcResult result = mockMvc.perform(get("/cart")
                        .contentType(MediaType.APPLICATION_JSON))
                .andExpect(status().isOk())
                .andReturn();
        ShoppingCartResponseDto actual = objectMapper.readValue(result.getResponse()
                .getContentAsString(), ShoppingCartResponseDto.class);

        assertNotNull(actual);
        assertEquals(expected, actual);
    }
```

Reply:
@WithMockUser creates a userdetails.User automatically, and the controller throws a ClassCastException because it needs a model.User. So my solution only solves the problem with this annotation)